Fw/Cfg/ConfigCheck.cpp

```diff
@@ -10,17 +10,18 @@
  *
  */
 
```

Comment:
With static asserts, can't we get rid of the ConfigCheck.cpp altogether?  I think you can put static asserts into the .hpp and not need a compilation unit just for assertions.

Reply:
We probably could, but you could already do that with the old system. The reason I thought this was in a cpp file was so that users could tweak FpConfig.hpp as they wished and during compilation we'd compile the ConfigCheck file to verify sure their config was valid. Is there somewhere else you would prefer to locate these checks?

Reply:
My point is if we put these at the bottom of the FpConfig.hpp header, the checks will happen when the header is included and this `.cpp`, which does nothing, will be removed.

Reply:
The only problem with leaving it in the header file is that users copy this header file into their own projects, moving the config validation out of the control of the F' framework itself.

Personally I like the pattern we currently have where the config is in a header file and the framework itself double checks that the provided config is valid.

I don't have a strong opinion - I'm happy to move these validation checks to the header if that's the pattern we want to move to.

Reply:
I see your point.  We can move-forward with how you have it and reevaluate later.